assets/js/state/sagas/hosts.js

```diff
@@ -52,6 +52,7 @@ export function* deregisterHost({ payload }) {
   yield put(setHostDeregistering(payload));
   try {
     yield call(del, `/hosts/${payload.id}`);
+    payload.navigate?.('/hosts');
```

Comment:
Never seen this format before.

Reply:
Fancy [Optional chaining with function calls](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls).

Basically, if `payload.navigate` doesn't exist, then when we try to call `payload.navigate()`, we instead just returned `undefined`, acting like a NOOP.

If we didn't do this, then when calling `payload.navigate()`, we would get the error thrown: `Uncaught TypeError: payload.navigate is not a function`

Reply:
We need to do this because we only want to navigate back to `HostsList` from `HostDetails`. It doesn't really make sense to do it from `HostsList` because we are already there and it is updated by websocket events. So as a result [we don't pass in `navigate`](https://github.com/trento-project/web/blob/c52a178fe0df9c39910cf31ced48c4141427544d/assets/js/components/HostsList.jsx#L68) to the payload to make this a NOOP.